### PR TITLE
Xml tooolkit issue195 remove construction from shade

### LINKED ISSUE #194: XML_Toolkit: WindowType present in TAS version
We should exclude `WindowType ` construction from TAS version

### LINKED ISSUE #195: XML_Toolkit: Remove Construction fro Shade TAS export
Remove constructionIdRef in `Shade `from TAS export
`    <Surface surfaceType="Shade" exposedToSun="false" constructionIdRef="B97115105993287971081083258328373779569888495837268" id="Panel-424">`

### LINKED ISSUE #196: XML_Toolkit: Remove transforming GLZ Elements into Curtain Element in TAS export
In TAS export we should not make modification to types and keep all elements as they are. We do all manipulation directly in TBD. Therefore Glazed floor should be kept as `ExposedFloor`

so this is error line in SAM template
`      <CADObjectId>Curtain Floor : SIM_EXT_GLZ_FLR Exposed FG01 [1012540]</CADObjectId>`

same apply to Glazed Roof.. we keep as `Roof`
`      <CADObjectId>Curtain Basic Roof : SIM_EXT_GLZ_Roof DG01 [722525]</CADObjectId>`

### PR DESCRIPTION
**Pull request by michaldengusiak**
### Issues addressed by this PR

Fixes #194
Fixes #195 
Fixes #196 

 
Fixes Tas version of BHoM gbXML export

 ### Test files

It is working on [sample file ](https://github.com/BHoM/samples/tree/master/gbXML_Toolkit)

 ### Changelog
#### Fixed 
 - Fixed Construction for Shades, Windows and transformation of GLZ object in TAS version of export

### CODE REVIEW

File: XML_Adapter/Serialise/BuildingElement.cs [hunk 1]
```diff
@@ -179,7 +179,11 @@ public static void SerializeCollection(IEnumerable<IEnumerable<BuildingElement>>
             gbx.Construction = usedConstructions.ToArray();
             gbx.Layer = usedLayers.ToArray();
             gbx.Material = usedMaterials.ToArray();
-            gbx.WindowType = usedWindows.ToArray();
+
+            if (exportType == ExportType.gbXMLIES)
+                gbx.WindowType = usedWindows.ToArray();
+            else if (exportType == ExportType.gbXMLTAS)//We have to force nill otherwise WindowType will be created
```
Comment (FraserGreenroyd) [inline XML_Adapter/Serialise/BuildingElement.cs]: Change `nill` to `null`

File: XML_Adapter/Serialise/BuildingElement.cs [hunk 2]
```diff
@@ -198,7 +202,10 @@ public static void SerializeCollection(IEnumerable<BuildingElement> inputElement
                 if (be.BuildingElementProperties != null)
                     gbSrf.CADObjectID = be.CADObjectID();
 
-                gbSrf.ConstructionIDRef = be.ConstructionID();
+                if (exportType == ExportType.gbXMLIES)
+                    gbSrf.ConstructionIDRef = be.ConstructionID();
+                else if (exportType == ExportType.gbXMLTAS) //We have to force nill otherwise Construction will be created
```
Comment (FraserGreenroyd) [inline XML_Adapter/Serialise/BuildingElement.cs]: Change `nill` to `null`

File: XML_Adapter/Serialise/BuildingElement.cs (hunk 1 above)
Comment (michaldengusiak) [reply XML_Adapter/Serialise/BuildingElement.cs]: done

File: XML_Adapter/Serialise/BuildingElement.cs (hunk 2 above)
Comment (michaldengusiak) [reply XML_Adapter/Serialise/BuildingElement.cs]: done,

**Comment by michaldengusiak** [comment]
#196  works fine it was wrong Source file I commited updated version

**Pull request merged**